### DISCUSSION TITLE
Example Creation Session - August 2024

### DISCUSSION THREAD

**Discussion by NicolaCourtier** [pybop-team/PyBOP#453] [Ideas]
We would like users and developers to come together to create more example notebooks and scripts for PyBOP. The aim is to broaden impact, catch current issues/bugs, and to collaborate on PyBOP usage.

If you have an idea for an example, please share it with us!

**Comment by NicolaCourtier** [comment]
One to start us off: #452 - add an example notebook on the `MultiFittingProblem` class based on the `multi_fitting` script.

**Comment by BradyPlanden** [comment]
A few suggestions for people to think on:

- ~~#437~~
- https://github.com/pybop-team/PyBOP/issues/282#issuecomment-2146037963
- #222
- #173
- Add a notebook for the recent transformation class: #357 
- Add a notebook for the recent weighted cost class: #329 
- Add an ECM MCMC sampling notebook from #340 
- Add example for new `cost.compute()` method #436

**Reply by Dibyendu-IITKGP** [reply]
A discussion on selection of solvers and how the solver time-step affects the fitting problem would be great.

**Reply by brosaplanella** [reply]
#222 is already assigned to me so happy to do that unless there's something more urgent :)

**Reply by agriyakhetarpal** [reply]
Picking up my work on #173, which is assigned to me.

**Reply by BradyPlanden** [reply]
I'm keen on #436

**Reply by NicolaCourtier** [reply]
I have a example here, ready for review: https://github.com/pybop-team/PyBOP/pull/442

**Comment by NicolaCourtier** [comment]
See this issue for previous discussion on restructuring the examples directory: https://github.com/pybop-team/PyBOP/issues/282

**Comment by BradyPlanden** [comment]
I've added the cost evaluation/compute notebook as: #470, I'll take on "Add an ECM MCMC sampling notebook from https://github.com/pybop-team/PyBOP/pull/340" now.
